src/lxml/tests/test_elementpath.py

```diff
@@ -285,6 +325,233 @@ def test_find(self):
         self.assertEqual(summarize_list(elem.findall(".//tag[@class][@id]")),
                          ['tag', 'tag'])
 
+    def test_find_with_namespaces(self):
+        elem = etree.XML(
+            """
+            <body xmlns="nsnone">
+              <tag class='a'>text</tag>
+              <tag class='b'/>
+              <section>
+                <tag class='b' id='inner'>subtext</tag>
+              </section>
+            </body>
+            """
+        )
+        namespaces = {None: "nsnone"}
+
+        self.assertEqual(elem.find("tag", namespaces=namespaces).tag, "{nsnone}tag")
+
+
+        self.assertEqual(
+            etree.ElementTree(elem).find("tag", namespaces=namespaces).tag, "{nsnone}tag"
+        )
+        self.assertEqual(elem.find("section/tag", namespaces=namespaces).tag, "{nsnone}tag")
+        self.assertEqual(
+            etree.ElementTree(elem).find("section/tag", namespaces=namespaces).tag,
+            "{nsnone}tag",
+        )
+        self.assertEqual(elem.findtext("tag", namespaces=namespaces), "text")
+        self.assertEqual(elem.findtext("tog", namespaces=namespaces), None)
+        self.assertEqual(elem.findtext("tog", "default", namespaces=namespaces), "default")
+        self.assertEqual(
+            etree.ElementTree(elem).findtext("tag", namespaces=namespaces), "text"
+        )
+        self.assertEqual(
+            etree.ElementTree(elem).findtext(
+                "tog", default="default", namespaces=namespaces
+            ),
+            "default",
+        )
+        self.assertEqual(elem.findtext("section/tag", namespaces=namespaces), "subtext")
+        self.assertEqual(
+            etree.ElementTree(elem).findtext("section/tag", namespaces=namespaces),
+            "subtext",
+        )
+        self.assertEqual(elem.find("tag[1]", namespaces=namespaces).tag, "{nsnone}tag")
+        self.assertEqual(
+            etree.ElementTree(elem).find("tag[1]", namespaces=namespaces).tag, "{nsnone}tag"
+        )
+
+        self.assertEqual(elem.findtext("tag[1]", namespaces=namespaces), "text")
+        self.assertTrue(elem.findtext("tag[2]", namespaces=namespaces) in {None, ''})
+        self.assertEqual(
+            summarize_list(elem.findall("tag", namespaces=namespaces)),
+            ["{nsnone}tag", "{nsnone}tag"],
+        )
```

Comment:
Given how many examples you add here, this seems worth its own custom assert method: `.assertFindallEqual(element, path, expected, namespaces=None)`.

Also, do we actually need to add these tests? ISTM that we could get away with running the existing tests three times, once without namespaces dict, once with an empty one, and once with a non-empty one.